sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -45,12 +49,169 @@
                                             hpat_join_series_indexes)
 from sdc.datatypes.hpat_pandas_series_rolling_types import _hpat_pandas_series_rolling_init
 from sdc.datatypes.hpat_pandas_stringmethods_types import StringMethodsType
+from sdc.datatypes.hpat_pandas_getitem_types import SeriesGetitemAccessorType
 from sdc.hiframes.pd_series_ext import SeriesType
 from sdc.str_arr_ext import (StringArrayType, string_array_type, str_arr_is_na, str_arr_set_na,
                              num_total_chars, pre_alloc_string_array, cp_str_list_to_array)
 from sdc.utils import to_array, sdc_overload, sdc_overload_method, sdc_overload_attribute
 
 
+@sdc_overload(operator.getitem)
+def hpat_pandas_series_accessor_getitem(self, idx):
+    """
+    Pandas Series operator :attr:`pandas.Series.get` implementation
+    **Algorithm**: result = series[idx]
+
+    **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_static_getitem_series1
+
+    Parameters
+    ----------
+    series: :obj:`pandas.Series`
+           input series
+    idx: :obj:`int`, :obj:`slice` or :obj:`pandas.Series`
+        input index
+
+    Returns
+    -------
+    :class:`pandas.Series` or an element of the underneath type
+            object of :class:`pandas.Series`
+    """
+
+    _func_name = 'Operator getitem().'
+
+    if not isinstance(self, SeriesGetitemAccessorType):
+        return None
+
+    accessor = self.accessor.literal_value
+
+    if accessor == 'iloc':
+        if isinstance(idx, types.SliceType):
+            def hpat_pandas_series_iloc_slice_impl(self, idx):
+                result_data = self._series._data[idx]
+                result_index = self._series.index[idx]
+                return pandas.Series(result_data, result_index, self._series._name)
+
+            return hpat_pandas_series_iloc_slice_impl
+
+        if isinstance(idx, (int, types.Integer)):
+            def hpat_pandas_series_iloc_impl(self, idx):
+                return self._series._data[idx]
+
+            return hpat_pandas_series_iloc_impl
+
+        raise TypingError('{} The index must be an Number, Slice, String, List, Array or a callable.\
+                    Given: {}'.format(_func_name, idx))
+
+    if accessor == 'iat':
+        if isinstance(idx, (int, types.Integer)):
+            def hpat_pandas_series_iat_impl(self, idx):
+                return self._series._data[idx]
+
+            return hpat_pandas_series_iat_impl
+
+        raise TypingError('{} The index must be a Integer. Given: {}'.format(_func_name, idx))
+
+    if accessor == 'loc':
+        # Note: Loc return Series
+        # Note: Index 0 in slice not supported
+        index_is_none = (self.series.index is None or
+                         isinstance(self.series.index, numba.types.misc.NoneType))
+        if isinstance(idx, types.SliceType) and not index_is_none:
+            def hpat_pandas_series_getitem_idx_slice_impl(self, idx):
```

Comment:
Here is two options how this function could be implemented:

Serial:
```python
def hpat_pandas_series_getitem_idx_slice_impl(self, idx):
    start_position = 0
    stop_position = 0
    
    series = self._series
    index = series.index

    if idx.start <= idx.stop:
        start_cmp = lambda indx, strt: indx < strt
        stop_cmp = lambda indx, stp: indx >= stp
        mono_cmp = lambda diff: diff > 0
    else:
        start_cmp = lambda indx, strt: indx > strt
        stop_cmp = lambda indx, stp: indx <= stp
        mono_cmp = lambda diff: diff < 0

    while start_cmp(index[start_position], idx.start) and start_position < len(index):
        start_position += 1

    stop_position = start_position
    prev_value = index[stop_position]
    while stop_cmp(index[stop_position], idx.stop) and stop_position < len(index):
        if not mono_cmp(prev_value - index[stop_position]):
            raise ValueError("index must be monotonic increasing or decreasing")
        prev_value = index[stop_position]
        stop_position += 1

    if start_position >= len(index) or stop_position <= 0 or stop_position <= start_position:
        return pd.Series(data=[], index=[], name=series._name)
    
    return pandas.Series(data=series._data[start_position:stop_position], 
                         index=index[start_position:stop_position], 
                         name=series._name)
```

Potentially parallel:
```python
def hpat_pandas_series_getitem_idx_slice_impl(self, idx):
    series = self._series
    index = series.index

    start_position = len(index)
    stop_position = 0
    max_diff = 0
    min_diff = 0
    for i in prange(len(index)):
        if index[i] >= idx.start:
            start_position = min(start_position, i)
        if index[i] <= idx.stop:
            stop_position = max(stop_position, i)
        if i > 0:
            max_diff = max(max_diff, index[i] - index[i - 1])
            min_diff = min(min_diff, index[i] - index[i - 1])
    
    if max_diff*min_diff < 0:
        raise ValueError("index must be monotonic increasing or decreasing")

    if start_position >= len(index) or stop_position <= 0 or stop_position <= start_position:
        return pd.Series(data=[], index=[], name=series._name)
    
    return pandas.Series(data=series._data[start_position:stop_position], 
                         index=index[start_position:stop_position], 
                         name=series._name)
```

Please note, I haven't verified either of them, so it may contain bugs.
Also behavior of parallel version is slightly different from behavior of serial one